docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,509 @@
+
+1.  Find an image you'd like to display when your program runs.
+    We used [this image of a computer with "hello world" on the screen](https://unsplash.com/photos/a-laptop-computer-sitting-on-top-of-a-wooden-desk-8q6e5hu3Ilc).
+    Save the image to your computer.
+
+1.  Create a test script on your computer and copy the following code into it:
+
+    {{< tabs >}}
+    {{% tab name="Python" %}}
+
+```python {class="line-numbers linkable-line-numbers"}
+# test.py opens an image and prints a random number
+from PIL import Image
+import random
+
+# TODO: Replace path with path to where you saved your photo
+photo = Image.open("/Users/jessamyt/Downloads/hello-world.jpg")
+
+photo.show()
+
+number = random.random()
+
+print("Hello, World! The latest random number is ", number, ".")
+```
+
+{{% /tab %}}
+
+<!-- TODO add back in when generator supports Go
+
+```go {class="line-numbers linkable-line-numbers"}
+// test.go opens an image and prints a random number
+package main
+
+import "fmt"
+import "math/rand"
+
+func main() {
+
+  number := rand.Float64()
+  fmt.Println("Hello, World! The latest random number is ", number, ".")
+
+  // TODO
+}
+```
+
+-->
+
+{{< /tabs >}}
+
+1.  Replace the path in the script above with the path to where you saved your photo.
+    Save the script.
+
+1.  Run the test script in your terminal:
+
+    {{< tabs >}}
+    {{% tab name="Python" %}}
+
+It's best practice to use a virtual environment for running Python scripts.
+You'll also need to install the dependency Pillow in the virtual environment before running the test script.
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+python3 -m venv .venv
+source .venv/bin/activate
+pip install Pillow
+python3 test.py
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+    The image you saved should open on your screen, and a random number should print to your terminal.
+
+1.  In later steps, the module generator will create a new virtual environment with required dependencies, so you can deactivate the one you just ran the test script in:
+
+    ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+    deactivate
+    ```
+
+## Choose an API to implement
+
+Now it's time to decide which Viam [APIs](/appendix/apis/#component-apis) make sense for your module.
+You need a way to return an image, and you need a way to return a number.
+
+If you look at the [camera API](/appendix/apis/components/camera/), you can see the `GetImage` method, which returns an image.
+That will work for the image.
+None of the camera API methods return a number though.
+
+Look at the [sensor API](/appendix/apis/components/sensor/), which includes the `GetReadings` method.
+You can return a number with that, but the sensor API can't return an image.
+
+Your module can provide multiple modular resources, so let's make two modular resources: a camera to return the image, and a sensor to return a random number.
```

Comment:
In this case we're talking about the module more just as the container for modular resources. It provides those resources once the module is deployed. So if we're talking about it in the context of how it'll be used I think provide is fine, in the context of "we can create two modular resources", I think it's definitely "contain". 

If you look at the UI, that also uses "contained"
<img width="317" alt="Screenshot 2024-10-31 at 13 03 19" src="https://github.com/user-attachments/assets/77dd0c10-af0c-4cdb-8601-57f9ce525773">

```suggestion
Your module can contain multiple modular resources, so let's make two modular resources: a camera to return the image, and a sensor to return a random number.
```